cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/worker/ListAppsTask.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.worker;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.inject.Inject;
+import io.cdap.cdap.api.service.worker.RunnableTask;
+import io.cdap.cdap.api.service.worker.RunnableTaskContext;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.sourcecontrol.operationrunner.ListAppsContext;
+import io.cdap.cdap.sourcecontrol.operationrunner.RepositoryAppsResponse;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The task that lists applications found in linked repository.
+ */
+public class ListAppsTask extends SourceControlTask implements RunnableTask  {
+  private static final Gson GSON = new GsonBuilder().create();
+  private static final Logger LOG = LoggerFactory.getLogger(ListAppsTask.class);
```

Comment:
Wonder if it makes sense to move the `GSON` and `LOG` into `SourceControlTask`?

Reply:
Loggers should be as specific as possible, so having it here is fine.